sdk/python/feast/staging/staging_strategy.py

```diff
@@ -0,0 +1,269 @@
+import re
+from abc import ABC, ABCMeta, abstractmethod
+from enum import Enum
+from tempfile import TemporaryFile
+from typing import List
+from urllib.parse import ParseResult, urlparse
+
+import boto3
```

Comment:
We should probably make these conditional imports, and not force users to import them.